# ITMGT 25.03 EE
# Assignment 3A

This is a data-cleaning assignment and is the first of two parts. The second part (3B) will be given Monday or Tuesday next week. This notebook is uploaded to our Github repository along with a csv of fake, dirty data called **recweekdata.csv**.


The output of this assignment is to convert recweekdata.csv into processable data for statistics. Submit the following files through GitHub:
1. A cleaned recweekdata.csv called cleanrecweekdata.csv **(5 points)**
2. This notebook, containing the code (properly commented) that you used to clean the data **(5 points)**
3. An html file of summary statistics from each column called **stats.html**. The requirements for the summary statistics are described below. **(10 points)**

This entire assignment (3A) is worth 20 points.

Upload your output files to your Github repositories. Make sure to send an email to **joben.ilagan@gmail.com** once you're done.

This assignment is due on **Tuesday, July 9, 2019** at 12:00NN.

In [6]:
## Code goes here
import re

import csv    

In [7]:
recweek_data = list()

with open('recweekdata.csv') as csv_file:
    csv_reader = csv.reader(csv_file, delimiter=',')
    count_line = 0
    for row in csv_reader:
        if count_line == 0:
            pass
        else:
            recweek_data.append({"firstname":row[0],"lastname":row[1],"birthmonth":row[2],"birthday":row[3],"birthyear":row[4],"course":row[5],"phonenumber":row[6]})
        count_line += 1
        
recweek_data

[{'firstname': 'name1',
  'lastname': 'lastname1',
  'birthmonth': '11',
  'birthday': '26',
  'birthyear': '1998',
  'course': 'BS LM',
  'phonenumber': '+63 918 302 6210'},
 {'firstname': 'name2',
  'lastname': 'lastname2',
  'birthmonth': 'May',
  'birthday': '26',
  'birthyear': '2001',
  'course': 'Management',
  'phonenumber': '927 212 6276'},
 {'firstname': 'name3',
  'lastname': 'lastname3',
  'birthmonth': '5',
  'birthday': '16',
  'birthyear': '1997',
  'course': 'BS CTM',
  'phonenumber': '+63 918 253 7017'},
 {'firstname': 'name4',
  'lastname': 'lastname4',
  'birthmonth': 'Dec',
  'birthday': '9',
  'birthyear': '97',
  'course': 'BS ITE',
  'phonenumber': '918 714 4360'},
 {'firstname': 'name5',
  'lastname': 'lastname5',
  'birthmonth': 'Sep',
  'birthday': '5',
  'birthyear': '97',
  'course': 'BS CTM',
  'phonenumber': '+63 927 462 1674'},
 {'firstname': 'name6',
  'lastname': 'lastname6',
  'birthmonth': 'March',
  'birthday': '2',
  'birthyear': '1999',
  'course':

In [25]:
# birthmonth
dirty_birthmonth = [] #unfiltered data for birthmonth
for i in recweek_data:
    dirty_birthmonth.append(str(i["birthmonth"]))

clean_birth_month = [re.sub(r'(January|Jan)',r'1',month) for month in clean_birth_month]
clean_birth_month = [re.sub(r'(February|Feb)',r'2',month) for month in clean_birth_month]
clean_birth_month = [re.sub(r'(March|Mar)',r'3',month) for month in clean_birth_month]
clean_birth_month = [re.sub(r'(April|Apr)',r'4',month) for month in clean_birth_month]
clean_birth_month = [re.sub(r'(May)',r'5',month) for month in clean_birth_month]
clean_birth_month = [re.sub(r'(June|Jun)',r'6',month) for month in clean_birth_month]
clean_birth_month = [re.sub(r'(July|Jul)',r'7',month) for month in clean_birth_month]
clean_birth_month = [re.sub(r'(August|Aug)',r'8',month) for month in clean_birth_month]
clean_birth_month = [re.sub(r'(September|Sep)',r'9',month) for month in clean_birth_month]
clean_birth_month = [re.sub(r'(October|Oct)',r'10',month) for month in clean_birth_month]
clean_birth_month = [re.sub(r'(November|Nov)',r'11',month) for month in clean_birth_month]
clean_birth_month = [re.sub(r'(December|Dec)',r'12',month) for month in clean_birth_month]

clean_birth_month

['11',
 '5',
 '5',
 '12',
 '9',
 '3',
 '11',
 '8',
 '7',
 '9',
 '12',
 '5',
 '1',
 '1',
 '6',
 '4',
 '1',
 '4',
 '12',
 '3',
 '7',
 '9',
 '4',
 '10',
 '5',
 '12',
 '3',
 '11',
 '10',
 '5',
 '4',
 '7',
 '12',
 '10',
 '7',
 '12',
 '2',
 '8',
 '5',
 '2',
 '11',
 '9',
 '1',
 '2',
 '1',
 '12',
 '8',
 '4',
 '10',
 '3',
 '8',
 '3',
 '12',
 '7',
 '5',
 '9',
 '5',
 '4',
 '4',
 '8',
 '9',
 '12',
 '2',
 '11',
 '3',
 '9',
 '12',
 '11',
 '11',
 '5',
 '8',
 '12',
 '6',
 '10',
 '9',
 '2',
 '5',
 '9',
 '12',
 '8',
 '2',
 '11',
 '6',
 '2',
 '1',
 '11',
 '8',
 '1',
 '3',
 '3',
 '8',
 '12',
 '6',
 '11',
 '9',
 '3',
 '1',
 '1',
 '5',
 '5',
 '4',
 '8',
 '12',
 '7',
 '3',
 '8',
 '4',
 '5',
 '8',
 '1',
 '12',
 '5',
 '5',
 '8',
 '3',
 '7',
 '1',
 '2',
 '12',
 '2',
 '3',
 '3',
 '4',
 '2',
 '11',
 '6',
 '3',
 '6',
 '10',
 '7',
 '2',
 '9',
 '4',
 '5',
 '10',
 '10',
 '6',
 '2',
 '4',
 '9',
 '6',
 '12',
 '11',
 '10',
 '2',
 '9',
 '12',
 '11',
 '9',
 '8',
 '5',
 '3',
 '11',
 '8',
 '12',
 '4',
 '8',
 '1',
 '10',
 '1

In [16]:
#birthday
clean_birthday = list()
for i in recweek_data:
    clean_birthday.append(str(i["birthday"]))

In [17]:
clean_lastname = list()
for i in recweek_data:
    clean_lastname.append(str(i["lastname"]))

In [18]:
clean_firstname = list()
for i in recweek_data:
    clean_firstname.append(str(i["firstname"]))

In [26]:
# birthyear
dirty_birthyear = [] #unfiltered data for birthyear
for i in recweek_data:
    dirty_birthyear.append(str(i["birthyear"]))
    

clean_birth_year = [re.sub(r'(1997|97)',r'97',yr) for yr in clean_birth_year]
clean_birth_year = [re.sub(r'(1998|98)',r'98',yr) for yr in clean_birth_year]
clean_birth_year = [re.sub(r'(1999|99)',r'99',yr) for yr in clean_birth_year]
clean_birth_year = [re.sub(r'(2000|00)',r'00',yr) for yr in clean_birth_year]
clean_birth_year = [re.sub(r'(2001|01)',r'01',yr) for yr in clean_birth_year]

clean_birth_year

['98',
 '01',
 '97',
 '97',
 '97',
 '99',
 '98',
 '99',
 '97',
 '01',
 '98',
 '97',
 '99',
 '99',
 '98',
 '99',
 '97',
 '97',
 '97',
 '97',
 '99',
 '99',
 '97',
 '97',
 '97',
 '99',
 '97',
 '00',
 '99',
 '99',
 '99',
 '99',
 '99',
 '99',
 '01',
 '98',
 '00',
 '01',
 '97',
 '98',
 '99',
 '98',
 '99',
 '98',
 '97',
 '98',
 '99',
 '99',
 '97',
 '01',
 '98',
 '99',
 '00',
 '98',
 '98',
 '98',
 '01',
 '97',
 '98',
 '99',
 '98',
 '00',
 '00',
 '97',
 '97',
 '97',
 '97',
 '01',
 '98',
 '98',
 '99',
 '99',
 '98',
 '97',
 '98',
 '97',
 '00',
 '00',
 '00',
 '99',
 '97',
 '97',
 '00',
 '99',
 '99',
 '98',
 '98',
 '97',
 '98',
 '99',
 '97',
 '97',
 '98',
 '97',
 '01',
 '99',
 '97',
 '98',
 '98',
 '99',
 '97',
 '01',
 '00',
 '01',
 '99',
 '97',
 '01',
 '99',
 '98',
 '99',
 '01',
 '97',
 '97',
 '99',
 '98',
 '97',
 '97',
 '99',
 '97',
 '97',
 '00',
 '01',
 '99',
 '99',
 '98',
 '00',
 '00',
 '98',
 '98',
 '00',
 '99',
 '99',
 '99',
 '00',
 '99',
 '00',
 '98',
 '97',
 '99',
 '99',
 '98',
 '99',
 '99',

In [30]:
# course
dirty_course = [] #unfiltered data for course
for i in recweek_data:
    dirty_course.append(str(i["course"]))

clean_course = [re.sub(r'(Management|BS MGT|MGT)',r'MGT',course) for course in dirty_course]
clean_course = [re.sub(r'(Information Technology Entrepreneurship|BS ITE|ITE)',r'ITE',course) for course in clean_course]
clean_course = [re.sub(r'(Management Engineering|MGT Engineering|BS ME|ME)',r'ME',course) for course in clean_course]
clean_course = [re.sub(r'(Legal Management|Legal MGT|BS LM|LM)',r'LM',course) for course in clean_course]
clean_course = [re.sub(r'(Communications Technology|BS CTM|CTM)',r'CTM',course) for course in clean_course]

clean_course

['LM',
 'MGT',
 'CTM',
 'ITE',
 'CTM',
 'MGT',
 'LM',
 'ME',
 'ME',
 'CTM',
 'LM',
 'ME',
 'CTM',
 'MGT',
 'ITE',
 'LM',
 'ITE',
 'MGT',
 'ITE',
 'ME',
 'LM',
 'ITE',
 'MGT',
 'ME',
 'ME',
 'ME',
 'CTM',
 'LM',
 'LM',
 'CTM',
 'ITE',
 'MGT',
 'ITE',
 'LM',
 'ITE',
 'MGT',
 'MGT',
 'LM',
 'ME',
 'LM',
 'MGT',
 'LM',
 'MGT',
 'ITE',
 'ME',
 'ME',
 'MGT',
 'ITE',
 'LM',
 'LM',
 'ITE',
 'CTM',
 'MGT',
 'CTM',
 'ME',
 'ME',
 'CTM',
 'ME',
 'ITE',
 'ITE',
 'CTM',
 'ITE',
 'ME',
 'ITE',
 'ME',
 'LM',
 'CTM',
 'MGT',
 'CTM',
 'LM',
 'CTM',
 'MGT',
 'ME',
 'LM',
 'ITE',
 'CTM',
 'CTM',
 'MGT',
 'LM',
 'CTM',
 'CTM',
 'CTM',
 'ME',
 'ITE',
 'LM',
 'MGT',
 'ITE',
 'ME',
 'ITE',
 'ITE',
 'ITE',
 'ME',
 'ITE',
 'LM',
 'CTM',
 'CTM',
 'MGT',
 'ITE',
 'CTM',
 'MGT',
 'CTM',
 'LM',
 'ME',
 'CTM',
 'ITE',
 'CTM',
 'MGT',
 'MGT',
 'ME',
 'ME',
 'ME',
 'MGT',
 'ITE',
 'CTM',
 'ITE',
 'ITE',
 'LM',
 'ME',
 'LM',
 'LM',
 'LM',
 'LM',
 'LM',
 'MGT',
 'LM',
 'LM',
 'LM',
 'LM',
 'ME',
 'MGT',
 'CTM',
 'LM',


In [31]:
# phone number
dirty_number = [] #unfiltered data for phone number
for i in recweek_data:
    dirty_number.append(str(i["phonenumber"]))

clean_number = [re.sub(r'\W+63 ',r'',number,1) for number in dirty_number]
clean_number = [re.sub(r'9', r'09',number,1) for number in clean_number]

clean_number

['0918 302 6210',
 '0927 212 6276',
 '0918 253 7017',
 '0918 714 4360',
 '0927 462 1674',
 '0918 629 5268',
 '0917 369 7388',
 '0917 240 6000',
 '0918 226 5748',
 '0918 342 1717',
 '0927 708 1308',
 '0927 747 5497',
 '0917 959 4490',
 '0927 543 4192',
 '0917 211 9497',
 '0918 185 5162',
 '0927 146 8146',
 '0927 516 4280',
 '0927 958 6909',
 '0917 514 9105',
 '0927 984 7499',
 '0927 699 3694',
 '0917 317 2478',
 '0917 159 3829',
 '0918 898 6972',
 '0917 796 4622',
 '0927 138 7580',
 '0917 195 9244',
 '0917 643 3996',
 '0917 758 3119',
 '0918 669 4265',
 '0917 750 8539',
 '0927 236 9504',
 '0917 438 4447',
 '0917 773 6035',
 '0918 455 4185',
 '0927 889 9252',
 '0927 874 8083',
 '0927 842 3053',
 '0927 923 1234',
 '0918 381 4956',
 '0918 602 9278',
 '0918 532 6748',
 '0917 654 1291',
 '0918 244 1683',
 '0917 818 1190',
 '0917 468 6425',
 '0918 635 1688',
 '0917 821 1896',
 '0927 924 9506',
 '0918 854 2626',
 '0918 438 9745',
 '0917 893 8235',
 '0918 216 5250',
 '0917 257 1057',
 '0927 949

In [34]:
clean_data = list()

for i in range (len(clean_number)):
    clean_data.append({"firstname":clean_firstname[i],"lastname":clean_lastname[i],"birthmonth":clean_birth_month[i],"birthday":clean_birthday[i],"birthyear":clean_birth_year[i],"course":clean_course[i],"phonenumber":clean_number[i]})
       
print(clean_data)

[{'firstname': 'name1', 'lastname': 'lastname1', 'birthmonth': '11', 'birthday': '26', 'birthyear': '98', 'course': 'LM', 'phonenumber': '0918 302 6210'}, {'firstname': 'name2', 'lastname': 'lastname2', 'birthmonth': '5', 'birthday': '26', 'birthyear': '01', 'course': 'MGT', 'phonenumber': '0927 212 6276'}, {'firstname': 'name3', 'lastname': 'lastname3', 'birthmonth': '5', 'birthday': '16', 'birthyear': '97', 'course': 'CTM', 'phonenumber': '0918 253 7017'}, {'firstname': 'name4', 'lastname': 'lastname4', 'birthmonth': '12', 'birthday': '9', 'birthyear': '97', 'course': 'ITE', 'phonenumber': '0918 714 4360'}, {'firstname': 'name5', 'lastname': 'lastname5', 'birthmonth': '9', 'birthday': '5', 'birthyear': '97', 'course': 'CTM', 'phonenumber': '0927 462 1674'}, {'firstname': 'name6', 'lastname': 'lastname6', 'birthmonth': '3', 'birthday': '2', 'birthyear': '99', 'course': 'MGT', 'phonenumber': '0918 629 5268'}, {'firstname': 'name7', 'lastname': 'lastname7', 'birthmonth': '11', 'birthday

In [35]:
with open('clean_recweek_data.csv', 'w+', newline='') as csvfile:
    fieldnames = ['firstname','lastname','birthmonth','birthday','birthyear','course','phonenumber']
    writer = csv.DictWriter(csvfile, fieldnames =fieldnames)
    writer.writeheader()
    for i in range(len(clean_firstname)):
        writer.writerow({'firstname':clean_data[i]['firstname'],
                         'lastname':clean_data[i]['lastname'],
                         'birthmonth':clean_data[i]['birthmonth'],
                         'birthday':clean_data[i]['birthday'],
                         'birthyear':clean_data[i]['birthyear'],
                         'course':clean_data[i]['course'],
                         'phonenumber':clean_data[i]['phonenumber']})

# Required Summary Statistics
1. How many people were born in each month?
2. How many people were born on each day?
3. How many people were born in each year?
4. How many people come from each course?
5. How many people have numbers from Globe (0917 or 0927) and Smart (0918)? *Note: it's up to you how to format this output*

Answer these questions in an html file called stats.html. The format of the html is as follows (in template form, so you can use this as format string in your program, though feel free to modify with variable names if you find it easier to do so):


```
<html>
    <body>
         <h2>Summary Statistics</h2>
         <table>
             <tr><th>Stat</th><th>Value</th></tr>
             <tr><td>People born in each month</td><td>{}</td></tr>
             <tr><td>People born on each day</td><td>{}</td></tr>
             <tr><td>People born in each year</td><td>{}</td></tr>
             <tr><td>People coming from each course</td><td>{}</td></tr>
             <tr><td>People having numbers from Globe and Smart</td><td>{}</td></tr>
         </table>
    </body>
</html>
```

In [40]:
summary_globe=0
summary_smart=0

for i in clean_number:
    if '0917' in i:
        summary_globe += 1
    elif '0927' in i:
        summary_globe +=1
    else:
        summary_smart += 1

In [41]:
from IPython.core.display import HTML

html = """
<h1>Summary Statistics</h1>
<table>
<tbody>
<tr>
<td>People born in each month</td>
<td>
<table>
<tbody>
<tr>
<th>January</th>
<th>February</th>
<th>March</th>
<th>April</th>
<th>May</th>
<th>June</th>
<th>July</th>
<th>August</th>
<th>September</th>
<th>October</th>
<th>November</th>
<th>December</th>
</tr>
<tr>
<td>{january}</td>
<td>{february}</td>
<td>{march}</td>
<td>{april}</td>
<td>{may}</td>
<td>{june}</td>
<td>{july}</td>
<td>{august}</td>
<td>{september}</td>
<td>{october}</td>
<td>{november}</td>
<td>{december}</td>
</tr>
</tbody>
</table>
</td>
</tr>
<tr>
<td>People born on each day</td>
<td>
<table>
<tbody>
<tr>
<th>1</th>
<th>2</th>
<th>3</th>
<th>4</th>
<th>5</th>
<th>6</th>
<th>7</th>
<th>8</th>
<th>9</th>
<th>10</th>
<th>11</th>
<th>12</th>
<th>13</th>
<th>14</th>
<th>15</th>
<th>16</th>
<th>17</th>
<th>18</th>
<th>19</th>
<th>20</th>
<th>21</th>
<th>22</th>
<th>23</th>
<th>24</th>
<th>25</th>
<th>26</th>
<th>27</th>
<th>28</th>
<th>29</th>
<th>30</th>
<th>31</th>
</tr>
<tr>
<td>{summary1}</td>
<td>{summary2}</td>
<td>{summary3}</td>
<td>{summary4}</td>
<td>{summary5}</td>
<td>{summary6}</td>
<td>{summary7}</td>
<td>{summary8}</td>
<td>{summary9}</td>
<td>{summary10}</td>
<td>{summary11}</td>
<td>{summary12}</td>
<td>{summary13}</td>
<td>{summary14}</td>
<td>{summary15}</td>
<td>{summary16}</td>
<td>{summary17}</td>
<td>{summary18}</td>
<td>{summary19}</td>
<td>{summary20}</td>
<td>{summary21}</td>
<td>{summary22}</td>
<td>{summary23}</td>
<td>{summary24}</td>
<td>{summary25}</td>
<td>{summary26}</td>
<td>{summary27}</td>
<td>{summary28}</td>
<td>{summary29}</td>
<td>{summary30}</td>
<td>{summary31}</td>
</tr>
</tbody>
</table>
</td>
</tr>
<tr>
<td>People born in each year</td>
<td>
<table>
<tbody>
<tr>
<th>1997</th>
<th>1998</th>
<th>1999</th>
<th>2000</th>
<th>2001</th>
</tr>
<tr>
<td>{summary97}</td>
<td>{summary98}</td>
<td>{summary99}</td>
<td>{summary00}</td>
<td>{summary01}</td>
</tr>
</tbody>
</table>
</td>
</tr>
<tr>
<td>People coming from each course</td>
<td>
<table>
<tbody>
<tr>
<th>BS Communications Technology Management</th>
<th>BS Information Technology Entrepreneurship</th>
<th>BS Legal Management</th>
<th>BS Management</th>
<th>BS Management Engineering</th>
</tr>
<tr>
<td>{summary_ctm}</td>
<td>{summary_ite}</td>
<td>{summary_lm}</td>
<td>{summary_mgt}</td>
<td>{summary_me}</td>
</tr>
</tbody>
</table>
</td>
</tr>
<tr>
<td>People having numbers from Globe and Smart</td>
<td>
<table>
<tbody>
<tr>
<th>Globe Users</th>
<th>Smart Users</th>
</tr>
<tr>
<td>{summary_globe}</td>
<td>{summary_smart}</td>
</tr>
</tbody>
</table>
</td>
</tr>
</tbody>
</table>

"""
html_output = html.format(january = clean_birth_month.count("1"),
                          february =clean_birth_month.count("2"),
                          march = clean_birth_month.count("3"),  
                          april = clean_birth_month.count("4"),
                          may = clean_birth_month.count("5"), 
                          june = clean_birth_month.count("6"),
                          july = clean_birth_month.count("7"),
                          august = clean_birth_month.count("8"),
                          september = clean_birth_month.count("9"),
                          october = clean_birth_month.count("10"),
                          november = clean_birth_month.count("11"),
                          december = clean_birth_month.count("12"),
                          summary1 = clean_birthday.count("1"),
                          summary2 = clean_birthday.count("2"),
                          summary3 = clean_birthday.count("3"),
                          summary4 = clean_birthday.count("4"),
                          summary5 = clean_birthday.count("5"),
                          summary6 = clean_birthday.count("6"),
                          summary7 = clean_birthday.count("7"),
                          summary8 = clean_birthday.count("8"),
                          summary9 = clean_birthday.count("9"),
                          summary10 = clean_birthday.count("10"),
                          summary11 = clean_birthday.count("11"), 
                          summary12 = clean_birthday.count("12"),
                          summary13 = clean_birthday.count("13"),
                          summary14 = clean_birthday.count("14"),
                          summary15 = clean_birthday.count("15"),
                          summary16 = clean_birthday.count("16"),
                          summary17 = clean_birthday.count("17"),
                          summary18 = clean_birthday.count("18"),
                          summary19 =clean_birthday.count("19"),
                          summary20 =clean_birthday.count("20"),
                          summary21 = clean_birthday.count("21"), 
                          summary22 = clean_birthday.count("22"), 
                          summary23 =clean_birthday.count("23"),
                          summary24 = clean_birthday.count("24"), 
                          summary25 = clean_birthday.count("25"),
                          summary26 = clean_birthday.count("26"), 
                          summary27 = clean_birthday.count("27"),
                          summary28 = clean_birthday.count("28"),
                          summary29 = clean_birthday.count("29"),
                          summary30 = clean_birthday.count("30"),
                          summary31 = clean_birthday.count("31"),
                          summary97 = clean_birth_year.count("97"),
                          summary98 = clean_birth_year.count("98"),
                          summary99 =clean_birth_year.count("99"),
                          summary00 =clean_birth_year.count("00"),
                          summary01 = clean_birth_year.count("01"),
                          summary_ctm = clean_course.count("CTM"),
                          summary_ite = clean_course.count("ITE"),
                          summary_lm = clean_course.count("LM"),
                          summary_mgt = clean_course.count("MGT"),
                          summary_me = clean_course.count("ME"),
                          summary_globe = summary_globe,
                          summary_smart = summary_smart
                         )
                          
                          
HTML(html_output)

with open('stats1.html',"w") as file:
    file.write(html_output)